### PR TITLE
Support emscripten on Windows

### PR DESCRIPTION
**Pull request by dschuff**
* Sync the windows node.js
* Build emscripten
* Fix test running and enable asm.js and emwasm tests

### CODE REVIEW

File: src/build.py [hunk 1]
```diff
@@ -545,7 +569,7 @@ def NoSync(*args):
     Source('cmake', '', '',  # The source and git args are ignored.
            custom_sync=SyncPrebuiltCMake),
     Source('nodejs', '', '',  # The source and git args are ignored.
-           custom_sync=SyncPrebuiltNodeJS, no_windows=True),
+           custom_sync=SyncPrebuiltNodeJS, no_windows=False),
```
Comment (jgravelle-google) [inline src/build.py]: Should just remove `no_windows=False`

File: src/build.py [hunk 2]
```diff
@@ -140,13 +142,14 @@ def Executable(name):
 
 
 def NodePlatformName():
-  if IsWindows():
-    return ''
-  return {'darwin': 'darwin-x64', 'linux2': 'linux-x64'}[sys.platform]
+  return {'darwin': 'darwin-x64',
+          'linux2': 'linux-x64',
+          'win32':'win32'}[sys.platform]
```
Comment (jgravelle-google) [inline src/build.py]: nit: space after `:` (flake8 doesn't care?)

File: src/build.py [hunk 3]
```diff
@@ -1170,7 +1195,7 @@ def AllBuilds(use_asm=False):
       Build('spec', Spec, no_windows=True),
       Build('binaryen', Binaryen),
       Build('fastcomp', Fastcomp),
-      Build('emscripten', Emscripten, True, use_asm),
+      Build('emscripten', Emscripten, False, use_asm),
```
Comment (jgravelle-google) [inline src/build.py]: We should pass these as kwargs

File: src/build.py (hunk 1 above)
Comment (dschuff) [inline src/build.py]: Done. Eventually I hope to remove `no_windows` entirely. Although it's likely we won't want to run every test or step on every platform so we may want something more general instead.

File: src/build.py (hunk 3 above)
Comment (dschuff) [inline src/build.py]: Done.

File: src/build.py (hunk 2 above)
Comment (dschuff) [inline src/build.py]: (already done, because flake8 does care)

**Pull request merged**